arbitrum-client/src/conversion.rs

```diff
@@ -141,6 +148,43 @@ pub fn to_contract_valid_wallet_update_statement(
     })
 }
 
+/// Convert a [`TransferAuxData`] to its corresponding smart contract type
+pub fn to_contract_transfer_aux_data(
+    data: &Option<TransferAuxData>,
+) -> Result<ContractTransferAuxData, ConversionError> {
+    let res = if let Some(data) = data {
+        match data {
+            TransferAuxData::Deposit(deposit) => ContractTransferAuxData {
+                permit_nonce: Some(
+                    AlloyU256::from_str(&biguint_to_hex_string(&deposit.permit_nonce))
+                        .map_err(|_| ConversionError::InvalidUint)?,
+                ),
+                permit_deadline: Some(
+                    AlloyU256::from_str(&biguint_to_hex_string(&deposit.permit_deadline))
+                        .map_err(|_| ConversionError::InvalidUint)?,
+                ),
+                permit_signature: Some(deposit.permit_signature.clone()),
+                transfer_signature: None,
+            },
+            TransferAuxData::Withdrawal(withdrawal) => ContractTransferAuxData {
```

Comment:
Why don't we just mimic the structure or even use the contracts type, don't see the need to separate out by withdrawals. Like I said elsewhere, we can group this with the external transfer struct and provide convenient constructors. These constructors could set fields to `None` as needed.

This would obviate the need for this logic which is sort of obscured by control flow. Plus the relayer never interprets this data anyways so we don't need much